libs/features/game-tele/src/edit-game-tele/game-tele.integration.spec.ts

```diff
@@ -0,0 +1,156 @@
+import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { RouterTestingModule } from '@angular/router/testing';
+import { GameTele } from '@keira/shared/acore-world-model';
+import { MysqlQueryService, SqliteService } from '@keira/shared/db-layer';
+import { EditorPageObject, TranslateTestingModule } from '@keira/shared/test-utils';
+import { ModalModule } from 'ngx-bootstrap/modal';
+import { ToastrModule } from 'ngx-toastr';
+import { of } from 'rxjs';
+import { instance, mock } from 'ts-mockito';
+import { GameTeleComponent } from './game-tele.component';
+import Spy = jasmine.Spy;
+import { GameTeleHandlerService } from '../game-tele-handler.service';
+
+class GameTelePage extends EditorPageObject<GameTeleComponent> {}
+
+describe('GameTele integration tests', () => {
+  let fixture: ComponentFixture<GameTeleComponent>;
+  let queryService: MysqlQueryService;
+  let querySpy: Spy;
+  let handlerService: GameTeleHandlerService;
+  let page: GameTelePage;
+
+  const id = 1;
+
+  const expectedFullCreateQuery =
+    'DELETE FROM `game_tele` WHERE (`id` = ' +
+    id +
+    ');\n' +
+    'INSERT INTO `game_tele` (`id`, `position_x`, `position_y`, `position_z`, `orientation`, `map`, `name`) VALUES\n' +
+    '(' +
+    id +
+    ", 0, 0, 0, 0, 0, '');\n";
+
+  const originalEntity = new GameTele();
+  originalEntity.id = 1;
+  originalEntity.name = 'ABC';
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        BrowserAnimationsModule,
+        ToastrModule.forRoot(),
+        ModalModule.forRoot(),
+        GameTeleComponent, // This should typically be in declarations, but as per your instruction, it's left unchanged
+        RouterTestingModule,
+        TranslateTestingModule,
+      ],
+      providers: [
+        GameTeleHandlerService,
+        {
+          provide: SqliteService,
+          useValue: instance(mock(SqliteService)),
+        },
+      ],
+    }).compileComponents();
+  }));
+
+  function setup(creatingNew: boolean) {
+    handlerService = TestBed.inject(GameTeleHandlerService);
+    handlerService['_selected'] = `${id}`;
+    handlerService.isNew = creatingNew;
+
+    queryService = TestBed.inject(MysqlQueryService);
+    querySpy = spyOn(queryService, 'query').and.returnValue(of([]));
+
+    spyOn(queryService, 'selectAll').and.returnValue(of(creatingNew ? [] : [originalEntity]));
+
+    fixture = TestBed.createComponent(GameTeleComponent);
+    page = new GameTelePage(fixture);
+    fixture.autoDetectChanges(true);
+    fixture.detectChanges();
+  }
+
+  describe('Creating new', () => {
+    beforeEach(() => setup(true));
+
+    it('should correctly initialise', () => {
```

Comment:
```suggestion
    it('should correctly initialise', () => {
      const { page } = setup(true);
```

and the same in the other `it` tests